packages/core/src/machines/actors/auth/resetPassword.ts

```diff
@@ -0,0 +1,115 @@
+import { createMachine, assign, sendUpdate } from 'xstate';
+
+import { AuthEvent, ResetPasswordContext } from '../../../types';
+import { Auth } from 'aws-amplify';
+
+export const resetPasswordActor = createMachine<
+  ResetPasswordContext,
+  AuthEvent
+>(
+  {
+    id: 'resetPasswordActor',
+    initial: 'resetPassword',
```

Comment:
I removed it in favor of `withContext` in the parent machine: https://github.com/aws-amplify/amplify-ui/blob/317c7cefb8b8e76e0620c7bdbac5ec006f808d1b/packages/core/src/machines/authMachine.ts#L141-L148.

Though I see the point of adding it in `resetPasswordMachine` too for readability and initialization purposes. I'll descope it out of this PR because it's already too big, but I'll add this to my TODO list.